cdbsearch.py

```diff
@@ -373,6 +377,24 @@ async def search(self, board, depth):
         if scored_db_moves == {}:
             return 0, ["invalid"]
 
+        # stop search if we are in EGTB and know the result
+        if (
+            not self.TBsearch
+            and sum(p in "pnbrqk" for p in board.epd().lower().split()[0]) <= CDB_EGTB
```

Comment:
I think this can be done without searching and string manipulation in the epd. For example this is counting code I have elsewhere:
```
                    knights = bin(board.knights).count("1")
                    bishops = bin(board.bishops).count("1")
                    rooks = bin(board.rooks).count("1")
                    queens = bin(board.queens).count("1")
                    pawns = bin(board.pawns).count("1")

                    matcountkey = (
                        9 * queens + 5 * rooks + 3 * knights + 3 * bishops + pawns
                    )
```

Reply:
Sure. But I do not think there is a big difference in efficiency here, and my code is 1 line and quite pythonic. :)

If you agree with the logic of this PR, we could leave non-functional changes for a follow-up PR?

Reply:
yes, can be done in a follow up.